packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -122,7 +128,48 @@ impl Abigen {
                            let provider = self.wallet.get_provider()?;
                            wallet.set_provider(provider.clone());
 
-                           Ok(Self { contract_id: self.contract_id.clone(), wallet: wallet })
+                           Ok(Self { contract_id: self.contract_id.clone(), wallet: wallet, logs_lookup: self.logs_lookup.clone() })
+                        }
+
+                        pub fn _logs_with_type<D: Tokenizable + Parameterize>(&self, receipts: &[Receipt]) -> Result<Vec<D>, SDKError> {
+                            let param_type = D::param_type();
```

Comment:
I'd extract everything from this method into a separate function in our codebase and delegate this call to it.

For example:
```rust
// in fuel-core for instance
pub fn something_ahmedm<T: Parameterize + Tokenizable>(
    logs_lookup: &[ResolvedLog],
    receipts: &[Receipt],
) -> Result<Vec<T>, Error> { // ...

// in the abigen
pub fn _logs_with_type<D: Tokenizable + Parameterize>(&self, receipts: &[Receipt]) -> Result<Vec<D>, SDKError> {
          something_ahmedm(&self.logs_lookup, receipts)
}
```

Because:
1. You'll get IDE support since this isn't part of a `quote!{}`
2. You can now unit test this logic.
3. If our clients decide to `abigen!` once and dump the result to a file, we'll still have control over the implementation, and porting bug fixes should be as easy as updating the SDK version.